### ISSUE TITLE
Implement multi-tenant auth

### ISSUE THREAD

**Issue by antkmsft** [Azure/azure-sdk-for-cpp#3688]
Multi-tenant auth is a superset of BearerTokenAuth: for some 401s we would get we can handle the multi-tenant auth by parsing `WWW-Authenticate`, and reauthenticate. Azure supports it.

**Comment by RickWinter** [comment]
At some point we will want to support this.  @antkmsft Can you provide details on who is requesting this work and the expected delivery.  Can you also add a cost estimate to the Project Cost field.

**Comment by antkmsft** [comment]
@RickWinter 
Currently, no customer is asking for this. It is just that recently I joined the meeting about this in other SDKs (this week, and there's a recording), and I have an idea how this can be implemented.

I think at one point George was implementing this, but the design was not in alignment with what other languages do, the PR (https://github.com/Azure/azure-sdk-for-cpp/pull/3365) was closed and I haven't seen a new one since.

I did put the cost in days here, on the right. If it is not showing up, I'll ask you how to do it, since this is the second time I am not putting it in the right place.
